go/fixchain/url_cache.go

```diff
@@ -20,7 +43,7 @@ type urlCache struct {
 }
 
 func (u *urlCache) getURL(url string) ([]byte, error) {
-	r, ok := u.cache[url]
+	r, ok := u.cache.get(url)
```

Comment:
FWIW, I did not make this locked in the first place because even if locked, a miss can still lead to a double fetch, unless you hold the lock for the whole fetch, which would be a bad idea.

That said, I suspect you need write locking to be thread safe (I forget!) - but it not, not sure there's much point in locking.